### ISSUE TITLE
Connect pull requests from other repos

### ISSUE THREAD

**Issue by adamralph** [waffleio/waffle.io#3349]
E.g. if a pull request from repo X is connected to an issue from repo Y, when looking at the board for repo Y, I would like to see the connected pull request in repo X against the issue in repo Y.

Note that I do _not_ want to add repo X to the board for repo Y. I do not want to see issues from repo X on the board for repo Y. I only want to see pull requests from repo X which are connected with issues from repo Y.

**Comment by burtonjc** [comment]
This is an interesting idea. I like it. Might be a bit complicated with permissions (ex. what if the viewer has access to Y but not X). We have mechanisms in place for this but they all assume that all issues displayed are from repos connected to your board. There would be similar complications around realtime updates.

What would you expect if you had a PR in repo Y that closed an issue in repo X? Would you want to see the issue from repo X with the PR from repo Y connected to it even though repo X is not part of your board?

**Comment by adamralph** [comment]
> What would you expect if you had a PR in repo Y that closed an issue in repo X? Would you want to see the issue from repo X with the PR from repo Y connected to it even though repo X is not part of your board?

Good question. For the specific case I have in mind, the reverse relationship in the direction you describe is not interesting, so I would be happy to see only the PR in repo Y.

However, if the reverse direction were implemented, then I guess it may even need a new UI, e.g. where we currently have this, meaning "PR 1740 in repo X connects to issue 1572 in repo Y (the current board)":

![image](https://user-images.githubusercontent.com/677704/34501925-69dd96aa-f011-11e7-9ca2-e72d5fed1dbe.png)

We may need something like this, to mean "PR 1572 in repo Y (the current board) connects to issue 1740 in repo X":

![image](https://user-images.githubusercontent.com/677704/34501847-0af77e8a-f011-11e7-9659-fa8d61b9c2b6.png)

Perhaps just the single direction (PRs from other repos connected to issues in the current board) would be a good first iteration?

**Comment by burtonjc** [comment]
Given our current architecture this would be a bit of a heavy lift. We currently do all of our associations out of our local cache of issue/pr data (which might not include a PR from a repo that Waffle does not know about). If we invest more heavily in GraphQL, we could possibly do some fancy stuff with [CrossReferencedEvent](https://developer.github.com/v4/object/crossreferencedevent/) on issue timelines to build those associations.

I really like this idea, it is most likely a ways out before we could do this though simply due to technical complexity.

**Comment by kurko** [comment]
Since 2013 (including the company I'm at right now), 95% of the companies and projects I worked at created a different repo for the product backlog because you usually have iOS, Android, Backend, Web repos separated, therefore one repo for product backlog makes sense.

In that context, I think this feature makes tons of sense.

**Comment by vasileiosC** [comment]
Agreed with the above, in a microservice oriented repository architecture where an organisation can have hundreds of repos (each of which can have its own set of recorded issues), including all of them in the product board for the sake of gaining PR visibility adds a lot of noise.
